anchore_engine/db/db_catalog_image.py

```diff
@@ -353,26 +351,31 @@ def delete(imageDigest, userId, session=None):
 
     return True
 
-#
-# def get_tags_by_time_range(session, userId, start=None, end=None):
-#     """
-#     Return the images with a created_at timestamp between the two parameters. Only returns the CatalogImageDocker record, not the source image.
-#
-#     :param session: active sqlachemy session
-#     :param userId:
-#     :param start:
-#     :param end:
-#     :return:
-#     """
-#
-#     if start and end:
-#         imgs = session.query(CatalogImageDocker).filter(CatalogImageDocker.userId == userId, CatalogImageDocker.created_at >= start,
-#                                                   CatalogImageDocker.created_at <= end).all()
-#     elif start:
-#         imgs = session.query(CatalogImageDocker).filter(CatalogImageDocker.userId == userId, CatalogImageDocker.created_at >= start).all()
-#     elif end:
-#         imgs = session.query(CatalogImageDocker).filter(CatalogImageDocker.userId == userId, CatalogImageDocker.created_at <= end).all()
-#     else:
-#         raise Exception('Must specify either start or end or both parameters')
-#
-#     return [x.to_dict() for x in imgs]
+
+def get_oldest_images_with_limit(session, max_images):
+    """
+    This method will return the oldest images which exceed the max_images limit
+
+    Ex. max_images = 1000
+        images_in_anchore = 1100
+
+    This method will return the image digests of the oldest 100 images (regardless of account)
+
+    :param session: DB Session
+    :param max_images: This is the maximum image count for the Anchore Deployment. Must be a positive integer.
+                       Based on the number of images in Anchore, this will determine the number returned images
+    """
+    if max_images < 0:
+        return
+
+    image_count = session.query(func.count()).order_by(None).scalar()
+    limit = image_count - max_images
+
+    if limit <= 0:
+        return
+
+    qry = session.query([CatalogImage.imageDigest])\
```

Comment:
Need to filter this by image status as well so you don't get analysis failed or analyzing images (depends on the input params that the user provides in the rule, but I think this should just be for 'analyzed' images.